addons/obiaway/Obiaway.lua

```diff
@@ -27,109 +27,136 @@
+require('logger')
 config = require('config')
 res = require('resources')
 
-defaults = {}
-defaults.location = 'sack'
+-- settings
+default_settings = {}
+default_settings.notify = true
+default_settings.location = 'sack'
+default_settings.lock = false
+settings = config.load(default_settings)
 
-settings = config.load(defaults)
+-- tokens
+tokens = {}
+tokens.lock_warned = false
+tokens.inv_full_warned = false
+tokens.bag_full_warned = false
 
-windower.register_event('addon command',function (...)
-    if args[1] == 'location' then
-        if S{'sack','case','satchel'}:contains(args[2]:lower()) then
-            settings.location = args[2]
-        else
-            print('Location can only be sack, satchel or case.')
-        end
-    else
-        remove_unneeded_obis()
-    end
-end)
+-- lists
+obi_cities = S{
+    "Ru'Lude Gardens",
+    "Upper Jeuno",
+    "Lower Jeuno",
+    "Port Jeuno",
+    "Port Windurst",
+    "Windurst Waters",
+    "Windurst Woods",
+    "Windurst Walls",
+    "Heavens Tower",
+    "Port San d'Oria",
+    "Northern San d'Oria",
+    "Southern San d'Oria",
+    "Port Bastok",
+    "Bastok Markets",
+    "Bastok Mines",
+    "Metalworks",
+    "Aht Urhgan Whitegate",
+    "Tavnazian Safehold",
+    "Nashmau",
+    "Selbina",
+    "Mhaura",
+    "Norg",
+    "Kazham",
+    "Eastern Adoulin",
+    "Western Adoulin",
+    "Leafallia",
+    "Celennia Memorial Library",
+    "Mog Garden"
+}
 
-function get_obis_in_inventory()
-    obis = {}
-    items = windower.ffxi.get_items()
-    inv = items.inventory
-    if not inv then return end
-    number = items.max_inventory
-    for i=1,number do 
-        id = inv[i].id
-        if ( id>=15435 and id<=15442) then
-            obis["Fire"] = obis["Fire"] or (id == 15435)
-            obis["Ice"] = obis["Ice"] or (id == 15436)
-            obis["Wind"] = obis["Wind"] or (id == 15437)
-            obis["Earth"] = obis["Earth"] or (id == 15438)
-            obis["Lightning"] = obis["Lightning"] or (id == 15439)
-            obis["Water"] = obis["Water"] or (id == 15440)
-            obis["Light"] = obis["Light"] or (id == 15441)
-            obis["Dark"] = obis["Dark"] or (id == 15442)
-        end
-    end
-    return obis
+obi_names = T{
+    Light = 'Korin',
+    Dark = 'Anrin',
+    Fire = 'Karin',
+    Earth = 'Dorin',
+    Water = 'Suirin',
+    Wind = 'Furin',
+    Ice = 'Hyorin',
+    Lightning = 'Rairin'
+}
+
+
+----------------------------------------
+---      UTITLITY Functions  
+----------------------------------------
+
+-- Automatically formats output text for addon. Accepts msg as a string.
+function obi_output(msg)
+    prefix = 'Obi: '
+    windower.add_to_chat(209, prefix..msg)
 end
 
-function remove_unneeded_obis()
-    elements = get_all_elements()
-    obis = get_obis_in_inventory()
+-- Pause function. Accepts a number of seconds to pause.
+function wait(seconds)
+  local start = os.time()
+  repeat until os.time() > start + seconds
+end
```

Comment:
You should remove this function and use `coroutine.sleep(time)` instead. It will not perform a busy wait like this code does (i.e. max your CPU usage).